### PR TITLE
Adopt Kotlin's binary compatibility validator

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by yschimke** [comment]
Nice - are these at HEAD? Should we go back to 4.9.3 and validate?

**Comment by swankjesse** [comment]
Closes: https://github.com/square/okhttp/issues/7032

**Comment by swankjesse** [comment]
Lemme backport to 4.9.3 manually and compare.

**Review by yschimke — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -241,6 +243,18 @@ subprojects {
...
+
+  plugins.withId("binary-compatibility-validator") {
+    configure<ApiValidationExtension> {
+      ignoredPackages += "okhttp3.logging.internal"
+      ignoredPackages += "mockwebserver3.internal"
+      ignoredPackages += "okhttp3.internal"
+      ignoredPackages += "mockwebserver3.junit5.internal"
+      ignoredPackages += "okhttp3.brotli.internal"
+      ignoredPackages += "okhttp3.sse.internal"
+      ignoredPackages += "okhttp3.tls.internal"
```
Comment (Goooler) [inline build.gradle.kts:255]: May use `+= arrayOf()`.

**Review by Goooler — APPROVED** [review]

**Comment by Goooler** [comment]
And add a validation job on CI? Like this:

https://github.com/coil-kt/coil/blob/936b35bce48109e01d650cbe6bb78a0e57e55111/.github/workflows/ci.yml#L35-L38

**Comment by Goooler** [comment]
Sorry, you are right, checkjdk11 job already contains it.

**Pull request merged**